src/service/template-impl.js

```diff
@@ -264,8 +264,11 @@ export class Templates {
   findTemplate(parent, opt_querySelector) {
     const templateElement = this.maybeFindTemplate(parent, opt_querySelector);
     user().assert(templateElement, 'Template not found for %s', parent);
-    user().assert(templateElement.tagName == 'TEMPLATE',
-        'Template element must be a "template" tag %s', templateElement);
+    const templateTagName = templateElement.tagName;
+    user().assert(
+        templateTagName == 'TEMPLATE' || templateTagName == 'SCRIPT',
```

Comment:
Assert script has `type="template"`

Reply:
the template tag also has a type that is enforced by the validator but isn't in the code base. 

e.g. `<template type="amp-mustache>..</template>`

Maybe it is enough to do that there and the script way of defining templates is going to follow the same model. I will do the validation changes in a separate PR.

Reply:
`<script type="template">` will need to be validated the same way a `<template>`. But, without this runtime check, I can pass a selector that will be a `<script type="text/json">`, which won't be validated. That would be a security issue.

Reply:
We should use a valid MIME type string to avoid future collisions, e.g. `text/html`. We can use a non-standard attribute `template="amp-mustache"` to denote template syntax.

https://html.spec.whatwg.org/multipage/scripting.html#attr-script-type